partition.go

```diff
@@ -0,0 +1,141 @@
+package schema
+
+import (
+	"encoding/binary"
+	"hash/fnv"
+	"sort"
+
+	"github.com/cespare/xxhash"
+	jump "github.com/dgryski/go-jump"
+)
+
+type PartitionByMethod uint8
+
+const (
+	PartitionByOrg PartitionByMethod = iota
+	PartitionBySeries
+	PartitionBySeriesWithTags
+	PartitionBySeriesWithTagsFnv
```

Comment:
the latter 2 could really benefit from a comment explaining their intended use (one of them for backwards compat, but not necessarily best distribution, the other one better distribution but non-backwards compat)

Reply:
added comments